### PR TITLE
[Feature] Legal hold simplify version

### PR DESCRIPTION
**Pull request by KaterinaWire**
## What's new in this PR?

This PR includes:
- block 1:1 conversations due to missing legal hold consent;
- new and updated clients should send `"capabilities": [ "legalhold-implicit-consent"]` to BE;
- add the property of participants removed reason to the system message.

### CODE REVIEW

File: Tests/Source/Model/User/ZMUserTests.m [hunk 1]
```diff
@@ -1362,6 +1362,25 @@ - (void)testThatIsBlockedIsTrueWhenThereIsABlockedConnection
...
+
+    // then
+    XCTAssertEqual(user.blockState, ZMBlockStateBlockedMissingLegalholdConsent);
+    XCTAssertTrue(user.isBlocked);
+
+    XCTAssertFalse(user.isConnected);
+    XCTAssertFalse(user.isIgnored);
+    XCTAssertFalse(user.isPendingApprovalByOtherUser);
+    XCTAssertFalse(user.isPendingApprovalBySelfUser);
+    XCTAssertTrue(user.canBeConnected);
```
Comment (johnxnguyen) [inline Tests/Source/Model/User/ZMUserTests.m:1382]: Is it correct that the user can be connected?

Comment (KaterinaWire) [reply Tests/Source/Model/User/ZMUserTests.m:1382]: Seems like yes. `canBeConnected` property does not depend on `isBlocked` property
```
- (BOOL)canBeConnected;
{
    if (self.isServiceUser || self.isWirelessUser) {
        return NO;
    }
    return ! self.isConnected && ! self.isPendingApprovalByOtherUser;
}
```

File: Source/Model/Message/ZMSystemMessage+ParticipantsRemovedReason.swift [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+            return nil
+        case .legalHoldPolicyConflict:
+            return "legalhold-policy-conflict"
+        @unknown default:
+            return nil
+        }
+    }
+
+    init(_ string: String) {
+        let allCases: [ZMParticipantsRemovedReason] = [.none, .legalHoldPolicyConflict]
```
Comment (David-Henner) [inline Source/Model/Message/ZMSystemMessage+ParticipantsRemovedReason.swift:34]: I don't know if `ZMParticipantsRemovedReason` can conform to `CaseIterable`, but if it does, you wouldn't need to set up the `allCases` variable yourself.

File: Source/Model/User/ZMUser.m [hunk 3]
```diff
@@ -871,14 +871,34 @@ @implementation ZMUser (Connections)
...
+            return ZMBlockStateBlockedMissingLegalholdConsent;
+        default:
+            return ZMBlockStateNone;
+    }
+}
+
++ (NSSet *)keyPathsForValuesAffectingBlockStateReason
+{
+    return [NSSet setWithObjects:ConnectionKey, @"connection.status", nil];
+}
```
Comment (David-Henner) [inline Source/Model/User/ZMUser.m:900]: isn't this the same as `keyPathsForValuesAffectingIsBlocked` ?

File: Source/Model/Message/ZMSystemMessage+ParticipantsRemovedReason.swift (hunk 2 above)
Comment (KaterinaWire) [reply Source/Model/Message/ZMSystemMessage+ParticipantsRemovedReason.swift:34]: `ZMParticipantsRemovedReason` is an objc Enum, so I needed to create `allCases`.

**Review by johnxnguyen — APPROVED** [review]

**Review by typfel — APPROVED** [review]

**Comment by KaterinaWire** [comment]
@zenkins release major

**Pull request merged**